lib/spack/spack/solver/concretize.lp

```diff
@@ -1186,7 +1186,8 @@ error(100, "{0} compiler '%{1}@{2}' incompatible with 'os={3}'", Package, Compil
      node_compiler(node(X, Package), CompilerID),
      compiler_name(CompilerID, Compiler),
      compiler_version(CompilerID, Version),
-     not compiler_os(CompilerID, OS),
+     compiler_os(CompilerID, CompilerOS),
+     not os_compatible(CompilerOS, OS)
```

Comment:
There's a syntax error:
```suggestion
     not os_compatible(CompilerOS, OS),
```

Reply:
Thanks! I copy pasted this into the web IDE. 🤦‍♂️